### DISCUSSION TITLE
SvelteKit 5 static site makes 3x network requests compared to equivalent Astro 4 static site

### DISCUSSION THREAD

**Discussion by sgoudham** [sveltejs/kit#13120] [Q&A]
Hey 👋,

I'm experimenting with moving my Astro 4 static site to a SvelteKit 5 static site as I've grown quite fond of Svelte's concepts enough to move the entire thing over. 

I've followed all the instructions listed in https://svelte.dev/docs/kit/adapter-static and have a successful build working. However, I'm noticing that the Svelte bundle is larger and also shipping `.js` chunks whereas the Astro one is not. The following screenshots are network requests of the homepage which has been ported over and has the same look/feel:

`Astro` (7 requests)
![image](https://github.com/user-attachments/assets/32b8137b-095e-457c-96aa-e9a57c7e77bf)

`SvelteKit` (21 requests)
![image](https://github.com/user-attachments/assets/7a0fda0c-e23d-44ac-8d86-e66f18df59ee)

From what I can see, Vite is still building the project as if SSR was enabled as it says so in the logs:

```
svelte-check found 0 errors and 5 warnings in 1 file
vite v6.0.3 building SSR bundle for production...
```

But I do know that the static adapter is being used:

```
Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-static
  Wrote site to "build"
  ✔ done
```

I'd greatly appreciate if someone is able to help me understand if this behaviour is intended, and if so, why?

Thanks!

**Comment by sgoudham** [comment]
Digging into the issues a little bit further, I think it's related to the following issues:
- https://github.com/sveltejs/kit/issues/9703
- https://github.com/sveltejs/kit/issues/12935

**Comment by mquandalle** [comment]
This is especially an issue when deploying on Vercel which bills based on the number of "edge requests" (uncached _and cached_).

I had to move my SvelteKit app out of Vercel because of this single issue. My app is doing about 100 HTTP requests per page, and the cost on Vercel just for that metric (ignoring CPU time, network usage, etc.) is $2.40 per 1,000,000 Requests, so $2.40 per 10,000 visits which gets expensive at the end of the month.

**Comment by bugproof** [comment]
This is also an issue with Cloudflare deploy I think. Can we expect any update on this this year? I really don't wanna use Astro. DX with SvelteKit is superior...
